### PR TITLE
Add NodeJS stack with react realworld app

### PR DESCRIPTION
**Pull request by tolusha**
Signed-off-by: Anatoliy Bazko <abazko@redhat.com>

### What does this PR do?
Adds NodeJS stack with the react realworld example [1]

### Issues
1. It is not possible to use `registry.access.redhat.com/ubi8/nodejs-10` image due to error `system limit for number of file watchers reached` when server starts (`npm start`).

2. `API_ROOT` [2] can't be modified due to issue [3]

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/13529

[1] https://github.com/gothinkster/react-redux-realworld-example-app
[2] https://github.com/gothinkster/react-redux-realworld-example-app/blob/master/src/agent.js#L6
[3] https://github.com/gothinkster/react-redux-realworld-example-app/issues/134

### CODE REVIEW

File: devfiles/nodejs-react/devfile.yaml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+components:
+  -
+    type: chePlugin
+    id: che-incubator/typescript/latest
+    memoryLimit: 512Mi
+  -
+    type: dockerimage
+    alias: nodejs
+    # system limit for number of file watchers reached with
+    # image: registry.access.redhat.com/ubi8/nodejs-10
```
Comment (rhopp) [inline devfiles/nodejs-react/devfile.yaml]: Is this issue logged/tracked somewhere?

File: devfiles/nodejs-react/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+---
+displayName: React web application
```
Comment (l0rd) [inline devfiles/nodejs-react/meta.yaml]: For consistency with other stacks I would rather rename it `NodeJS React Web Application`

File: devfiles/nodejs-react/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,6 @@
...
+description: NodeJS 10 stack with React real world application
```
Comment (l0rd) [inline devfiles/nodejs-react/meta.yaml]: Again for consistency: "Stack for developing NodeJS React Web Application"

File: devfiles/nodejs-react/devfile.yaml [hunk 4]
```diff
@@ -0,0 +1,48 @@
+---
+apiVersion: 1.0.0
+metadata:
+  name: nodejs-react
```
Comment (l0rd) [inline devfiles/nodejs-react/devfile.yaml]: For consistency with angular I would call it just `react`

File: devfiles/nodejs-react/devfile.yaml (hunk 1 above)
Comment (tolusha) [reply devfiles/nodejs-react/devfile.yaml]: @rhopp 
No. Should issue be created in eclipse/che repo?

**Pull request merged**